### ISSUE TITLE
Failure to check out revision pre-commit v1.15.1

### ISSUE THREAD

**Issue by asottile** [pre-commit/pre-commit#997]
```console
$ cat .pre-commit-config.yaml 
repos:
-   repo: https://github.com/asottile/pyupgrade
    rev: v1.13.0
    hooks:
    -   id: pyupgrade
$ pre-commit install-hooks
[INFO] Initializing environment for https://github.com/asottile/pyupgrade.
An unexpected error has occurred: CalledProcessError: Command: ('/usr/bin/git', 'checkout', 'v1.13.0')
Return code: 1
Expected return code: 0
Output: (none)
Errors: 
    error: pathspec 'v1.13.0' did not match any file(s) known to git.
    

Check the log at /code/.cache/pre-commit/pre-commit.log
lyftsvc@b780c05acb2e:/code/places$ cat ~/.cacp^C
lyftsvc@b780c05acb2e:/code/places$ cat /code/.cache/pre-commit/pre-commit.log 
An unexpected error has occurred: CalledProcessError: Command: ('/usr/bin/git', 'checkout', 'v1.13.0')
Return code: 1
Expected return code: 0
Output: (none)
Errors: 
    error: pathspec 'v1.13.0' did not match any file(s) known to git.
    

Traceback (most recent call last):
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 172, in clone_strategy
    self._shallow_clone(ref, _git_cmd)
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 149, in _shallow_clone
    git_cmd('-c', git_config, 'fetch', 'origin', ref, '--depth=1')
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 166, in _git_cmd
    cmd_output('git', *args, cwd=directory, env=env)
  File "/venv/lib/python3.6/site-packages/pre_commit/util.py", line 153, in cmd_output
    returncode, cmd, retcode, output=(stdout, stderr),
pre_commit.util.CalledProcessError: Command: ('/usr/bin/git', '-c', 'protocol.version=2', 'fetch', 'origin', 'v1.13.0', '--depth=1')
Return code: 128
Expected return code: 0
Output: (none)
Errors: 
    fatal: unknown value for config 'protocol.version': 2
    


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/venv/lib/python3.6/site-packages/pre_commit/error_handler.py", line 46, in error_handler
    yield
  File "/venv/lib/python3.6/site-packages/pre_commit/main.py", line 274, in main
    return install_hooks(args.config, store)
  File "/venv/lib/python3.6/site-packages/pre_commit/commands/install_uninstall.py", line 129, in install_hooks
    install_hook_envs(all_hooks(load_config(config_file), store), store)
  File "/venv/lib/python3.6/site-packages/pre_commit/repository.py", line 221, in all_hooks
    for repo in root_config['repos']
  File "/venv/lib/python3.6/site-packages/pre_commit/repository.py", line 222, in <genexpr>
    for hook in _repository_hooks(repo, store, root_config)
  File "/venv/lib/python3.6/site-packages/pre_commit/repository.py", line 197, in _repository_hooks
    return _cloned_repository_hooks(repo_config, store, root_config)
  File "/venv/lib/python3.6/site-packages/pre_commit/repository.py", line 166, in _cloned_repository_hooks
    manifest_path = os.path.join(store.clone(repo, rev), C.MANIFEST_FILE)
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 176, in clone
    return self._new_repo(repo, ref, deps, clone_strategy)
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 128, in _new_repo
    make_strategy(directory)
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 174, in clone_strategy
    self._complete_clone(ref, _git_cmd)
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 142, in _complete_clone
    git_cmd('checkout', ref)
  File "/venv/lib/python3.6/site-packages/pre_commit/store.py", line 166, in _git_cmd
    cmd_output('git', *args, cwd=directory, env=env)
  File "/venv/lib/python3.6/site-packages/pre_commit/util.py", line 153, in cmd_output
    returncode, cmd, retcode, output=(stdout, stderr),
pre_commit.util.CalledProcessError: Command: ('/usr/bin/git', 'checkout', 'v1.13.0')
Return code: 1
Expected return code: 0
Output: (none)
Errors: 
    error: pathspec 'v1.13.0' did not match any file(s) known to git.
```

Perhaps related to the new clone strategy

**Comment by asottile** [comment]
I just ~partially fixed this, but not the root cause -- v1.13.0 was tagged, but not on the mainline (I retagged the correct commit there)

the full-clone is not fetching all tags, the fix is to add `--tags` to the `fetch` line

**Comment by asottile** [comment]
CC @DanielChabrowski just fyi I misled you :laughing:
